frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,511 @@
+                className={`message-receiver-break-word ${checkForActiveColumn(
+                    ColumnDataEnum.TransportResults
+                )}`}
+                onClick={() => {
+                    setModalText({
+                        title: `${ColumnDataEnum.TransportResults}:`,
+                        body: receiver.TransportResults,
+                    });
+                    modalRef?.current?.toggleModal();
+                }}
+            >
+                {receiver.TransportResults}
+            </td>
+        </tr>
+    );
 };
 
 export const MessageReceivers = ({ receiverDetails }: MessageReceiverProps) => {
+    const modalRef = useRef<ModalRef>(null);
+    const [modalText, setModalText] = useState({ title: "", body: "" });
+    const [activeColumn, setActiveColumn] = useState("");
+    const [activeColumnSortOrder, setActiveColumnSortOrder] = useState(
+        FilterOptionsEnum.NONE
+    );
+    // Since we're splicing together both data from receiverDetails && fileUrl
+    // To simplify any proceeding logic within this component, we need to
+    // merge and normalize the data. The data currently appears like:
+    //   {
+    //     "reportId": "4b3c73df-83b1-48f9-a5a2-ce0c38662f7c",
+    //     "receivingOrg": "ignore",
+    //     "receivingOrgSvc": "HL7_NULL",
+    //     "transportResult": null,
+    //     "fileName": null,
+    //     "fileUrl": "http://azurite:10000/devstoreaccount1/reports/batch%2Fignore.HL7_NULL%2Ftx-covid-19-4b3c73df-83b1-48f9-a5a2-ce0c38662f7c-20230203182255.internal.csv",
+    //     "createdAt": "2023-02-03T18:22:55.580322",
+    //     "qualityFilters": []
+    // }
+    // And we parse fileUrl using parseFileLocation to find: folderLocation, sendingOrg and fileName. We use normalizedData to fix this. If there's no data, we fill in the string with N/A.
+
+    // Our expected output should look like the following:
+    //   NormalizedReceiverData {
+    //     Name: string;
+    //     Service: string;
+    //     Date: string;
+    //     ReportId: string;
+    //     Main: string;
+    //     Sub: string;
+    //     FileName: string;
+    //     TransportResults: string;
+    // }
+    const normalizedData = useMemo(
+        () =>
+            receiverDetails.map(
+                (receiverItem: ReceiverData): NormalizedReceiverData => {
+                    const formattedData = Object.keys(ColumnDataEnum).reduce(
+                        (accumulator: any, currentValue: string) => {
+                            accumulator[currentValue] = "N/A";
+                            return accumulator;
+                        },
+                        {}
+                    );
+                    for (const key of Object.keys(ColumnDataEnum)) {
+                        const columnTitle =
+                            ColumnDataEnum[key as keyof typeof ColumnDataEnum];
+                        switch (true) {
+                            case columnTitle === ColumnDataEnum.Name:
```

Comment:
Can this be simplified a bit?  Essentially, we have a few different groups of data mappers/normalizers:

- No-ops/1:1s: Name, Service, ReportId, TransportResults
- Date formatter: Date
- File url formatter: Main, Sub, FileName

We're always going to need to have a data mapper to transform response data for what we want to display, but I think we can cut down on some repeated code.